### PR TITLE
feat(manage): remove old agent and applicant contact journey

### PR DESCRIPTION
**Pull request by RobBroadhead30**
## Describe your changes
This pull request completes the migration to support multiple applicants and agents in the "create-a-case" flow, removing legacy logic and data structures related to single applicant/agent handling. The codebase is simplified by eliminating now-obsolete fields and functions, and the notification association logic is updated to resolve contacts through organisations rather than direct contact fields. Tests are updated and expanded to cover the new data structures and edge cases.

**Migration to multiple applicants and agents:**

* Removed all legacy logic and code paths for single applicant/agent handling, including `ApplicantContact` and `AgentContact` fields, the `createJourneyV2` function, and related SharePoint access and notification logic. All relevant flows now use the new data structures for managing multiple applicants and agents. [[1]](diffhunk://#diff-c9f32035ccc6643be09a45933eb06fac87dbfdb8444e50d858edafce397c2399L465-L496) [[2]](diffhunk://#diff-c9f32035ccc6643be09a45933eb06fac87dbfdb8444e50d858edafce397c2399L514-L527) [[3]](diffhunk://#diff-c9f32035ccc6643be09a45933eb06fac87dbfdb8444e50d858edafce397c2399L591-L619) [[4]](diffhunk://#diff-c9f32035ccc6643be09a45933eb06fac87dbfdb8444e50d858edafce397c2399L659-L683) [[5]](diffhunk://#diff-c9f32035ccc6643be09a45933eb06fac87dbfdb8444e50d858edafce397c2399L774-R672) [[6]](diffhunk://#diff-59c8a29473e33f26767d1a65173e58e5a5daadefec73c37db135773867c00b2aL12-R12) [[7]](diffhunk://#diff-59c8a29473e33f26767d1a65173e58e5a5daadefec73c37db135773867c00b2aL34-R34) [[8]](diffhunk://#diff-1b2787aa618a50684dfe770cd2b9d9867ea5560466fddb0988987ad90e4c7a94L20-L74)

* Updated the `extractAgentParty` function to require agent contact details when an agent is present, enforcing the new data model.

**Notification association improvements:**

* Refactored notification association logic to resolve contacts through the `Organisations` relationship and their associated contacts, removing reliance on direct `ApplicantContact` and `AgentContact` fields.

**Test updates and coverage:**

* Updated and expanded tests to use the new applicant/agent data structures, and removed tests for legacy code paths. Added new tests to ensure correct contact resolution with multiple organisations and fallback behavior. [[1]](diffhunk://#diff-982bfca62d3b4842deeabd3aba8b6b71c220df3d3602807c5ec870c653092c39L319-R320) [[2]](diffhunk://#diff-982bfca62d3b4842deeabd3aba8b6b71c220df3d3602807c5ec870c653092c39L355-R368) [[3]](diffhunk://#diff-982bfca62d3b4842deeabd3aba8b6b71c220df3d3602807c5ec870c653092c39L392-R416) [[4]](diffhunk://#diff-982bfca62d3b4842deeabd3aba8b6b71c220df3d3602807c5ec870c653092c39R529-R622) [[5]](diffhunk://#diff-28c6c581c96c1e8ea9d806b0d8caa77828d239fea9c7018502069e9ccdc45548L3-L12) [[6]](diffhunk://#diff-28c6c581c96c1e8ea9d806b0d8caa77828d239fea9c7018502069e9ccdc45548L45-R41) [[7]](diffhunk://#diff-28c6c581c96c1e8ea9d806b0d8caa77828d239fea9c7018502069e9ccdc45548L82-R87) [[8]](diffhunk://#diff-24230c01da5f0ebb8ce934d0140763c407e479382528ef894ccf2bcf83e5f0b1L168-R181) [[9]](diffhunk://#diff-24230c01da5f0ebb8ce934d0140763c407e479382528ef894ccf2bcf83e5f0b1L203-R220) [[10]](diffhunk://#diff-24230c01da5f0ebb8ce934d0140763c407e479382528ef894ccf2bcf83e5f0b1L278-R304)

These changes complete the transition to a more flexible and robust model for handling multiple applicants and agents, and clean up the codebase by removing all obsolete logic and tests.
## Issue ticket number and link
https://pins-ds.atlassian.net/browse/CROWN-1509

### CODE REVIEW

File: apps/manage/src/app/views/cases/create-a-case/types.d.ts
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/types.d.ts]: These are the old agent fields and need removing

File: apps/manage/src/app/views/cases/create-a-case/types.d.ts [hunk 1]
```diff
@@ -45,9 +40,8 @@ export interface CreateCaseAnswers {
 	expectedDateOfSubmission?: string;
 	agentOrganisationName?: string;
 	agentOrganisationAddress?: Address;
-	manageAgentContactDetails?: AgentContact[];
-	// TODO make this required when we fully switch over to V2 CROWN-1509
-	manageApplicantDetails?: ApplicantOrganisation[];
-	manageApplicantContactDetails?: ApplicantContact[];
+	manageAgentContactDetails: AgentContact[];
+	manageApplicantDetails: ApplicantOrganisation[];
+	manageApplicantContactDetails: ApplicantContact[];
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/types.d.ts]: Only `manageApplicantDetails` should be required - the other two fields can be optional depending on the value of `hasAgent`
```suggestion
	manageAgentContactDetails?: AgentContact[];
	manageApplicantDetails: ApplicantOrganisation[];
	manageApplicantContactDetails?: ApplicantContact[];
```

File: apps/manage/src/app/views/cases/create-a-case/save.js
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/save.js]: This should be renamed to lose the V2

Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/save.js]: Same here, rename to drop the V2

File: apps/manage/src/app/views/cases/create-a-case/save.js [hunk 2]
```diff
@@ -283,10 +283,8 @@ function extractAgentParty(answers) {
 		return null;
 	}
 
-	// For the old flow - hasAgent might be true without multiple agent contacts
-	// TODO - throw error after migration CROWN-1509
 	if (!hasAnswers(answers, 'manageAgentContactDetails')) {
-		return null;
+		throw new Error('Agent contacts are required when the case has an agent');
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/save.js:287]: Is there a test for this?

File: apps/manage/src/app/views/cases/view/controller.test.js [hunk 3]
```diff
@@ -126,34 +126,10 @@ describe('case details', () => {
 			await assert.doesNotReject(() => middleware(mockReq, mockRes, next));
 			assert.strictEqual(next.mock.callCount(), 1);
 			assert.ok(mockRes.locals.journey);
-			//Check that the journey is created using createJourneyV2 by checking for a property unique to V2 journeys
+			// Check that the journey includes multiple applicants support
 			assert.deepStrictEqual(JSON.stringify(mockRes.locals.journey).includes('manageApplicantDetails'), true);
 		});
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/controller.test.js]: This test is redundant - the code is no longer checking `isMultipleApplicantsLive`. You can remove this test and also look to remove the entire `isMultipleApplicantsLive` flag, since it is no longer being checked by any logic

File: apps/manage/src/app/views/cases/view/journey.js [hunk 4]
```diff
@@ -52,179 +52,6 @@ export function createJourney(questions, response, req) {
...
-				.addQuestion(questions.lpaContact)
-				.addQuestion(questions.lpaAddress)
-				.addQuestion(questions.secondaryLpaContact)
-				.withCondition(hasSecondaryLpa)
-				.addQuestion(questions.secondaryLpaAddress)
-				.withCondition(hasSecondaryLpa)
-				.addQuestion(questions.applicantContact)
-				.addQuestion(questions.applicantContactAddress)
-				.addQuestion(questions.agentContact)
-				.addQuestion(questions.agentContactAddress),
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/journey.js:107]: Now we're no longer using these `applicantContact` and `agentContact` questions, you can also remove them from `questions.js` and get rid of the `contactQuestions` helper

File: apps/manage/src/app/views/cases/create-a-case/journey.js [hunk 5]
```diff
@@ -17,61 +17,6 @@ export function createJourney(questions, response, req) {
...
-				.addQuestion(questions.agentEmail)
-				.withCondition(whenQuestionHasAnswer(questions.hasAgent, BOOLEAN_OPTIONS.YES))
-				.addQuestion(questions.agentTelephoneNumber)
-				.withCondition(whenQuestionHasAnswer(questions.hasAgent, BOOLEAN_OPTIONS.YES))
-				.addQuestion(questions.applicantEmail)
-				.withRequiredCondition(
-					whenQuestionHasAnswer(questions.hasAgent, BOOLEAN_OPTIONS.NO),
-					'Enter Applicant email address'
-				)
-				.addQuestion(questions.applicantTelephoneNumber)
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/journey.js:45]: Same as on view case - now we're no longer using these `applicantContact` and `agentContact` questions, you can also remove them from `questions.js` and get rid of the `contactQuestions` helper

File: apps/manage/src/app/views/cases/create-a-case/journey.test.js [hunk 6]
```diff
@@ -79,16 +75,16 @@ describe('create-a-case journey', () => {
 		assert.strictEqual(secondaryLpaQ.shouldDisplay(response), true);
 	});
 });
-describe('create-a-case journey V2', () => {
+describe('create-a-case journey ', () => {
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/create-a-case/journey.test.js]: You've got duplication with the same `describe` statement above, and the same `all questions should be defined` test - make sure to remove the unneeded tests instead of just renaming them

File: apps/manage/src/app/views/cases/view/publish/controller.test.js
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/publish/controller.test.js]: This test was testing for feature flag - it's now redundant, so remove

File: apps/manage/src/app/views/cases/view/journey.test.js
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/journey.test.js:25]: Same as the other journey tests - this is now duplicating the test on line 14

File: packages/database/src/schema.prisma [hunk 7]
```diff
@@ -52,14 +52,6 @@ model CrownDevelopment {
...
   Organisations            CrownDevelopmentToOrganisation[]
 
-  /// Applicant contact details (To be migrated to ApplicantContacts)
-  ///TODO: Remove applicantContactId and ApplicantContact relation after migration CROWN-1509
-  applicantContactId                  String?                     @db.UniqueIdentifier
-  ApplicantContact                    Contact?                    @relation("ApplicantContact", fields: [applicantContactId], references: [id], onUpdate: NoAction, onDelete: NoAction)
-  /// Agent contact details (To be migrated to AgentContacts)
-  ///TODO: Remove agentContactId and AgentContact relation after migration CROWN-1509
-  agentContactId                      String?                     @db.UniqueIdentifier
-  AgentContact                        Contact?                    @relation("AgentContact", fields: [agentContactId], references: [id], onUpdate: NoAction, onDelete: NoAction)
```
Comment (pinssbrett) [inline packages/database/src/schema.prisma:62]: Do we have a migration plan for the existing data in the production database?

File: apps/manage/src/app/views/cases/view/update-case.test.js [hunk 8]
```diff
@@ -1033,20 +1033,38 @@ describe('case details', () => {
...
+										{
+											Contact: { id: 'c1', email: 'test@email.com', firstName: 'Test', lastName: 'User' }
+										}
+									]
+								}
+							}
+						],
+						ApplicantContact: null,
+						AgentContact: null,
+						agentContactId: null
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/update-case.test.js]: No need to include these keys now
```suggestion
```

File: apps/manage/src/app/views/cases/view/update-case.test.js [hunk 9]
```diff
@@ -1033,20 +1033,38 @@ describe('case details', () => {
...
 			const mockNotifyClient = {
-				sendApplicationReceivedNotification: mock.fn()
+				sendApplicationReceivedNotificationToMany: mock.fn()
 			};
 			const updateCase = buildUpdateCase({
 				db: mockDb,
 				logger,
-				notifyClient: mockNotifyClient
+				notifyClient: mockNotifyClient,
+				isMultipleApplicantsLive: true
```
Comment (pinssbrett) [inline apps/manage/src/app/views/cases/view/update-case.test.js]: No need to test this if it's being removed
```suggestion
```

File: packages/database/src/schema.prisma [hunk 10]
```diff
@@ -52,14 +52,6 @@ model CrownDevelopment {
   /// Applicant organisations details
   Organisations            CrownDevelopmentToOrganisation[]
 
-  /// Applicant contact details (To be migrated to ApplicantContacts)
-  ///TODO: Remove applicantContactId and ApplicantContact relation after migration CROWN-1509
-  applicantContactId                  String?                     @db.UniqueIdentifier
-  ApplicantContact                    Contact?                    @relation("ApplicantContact", fields: [applicantContactId], references: [id], onUpdate: NoAction, onDelete: NoAction)
```
Comment (pinssbrett) [inline packages/database/src/schema.prisma:58]: `ApplicantContact` and `AgentContact` are still used in a few tests:

- `apps/portal/src/app/views/applications/list/controller.test.js`
- `apps/manage/src/app/views/cases/view/update-case.test.js`
- `apps/portal/src/app/views/applications/view/application-info/controller.test.js`

File: apps/manage/src/app/notify/controller.js [hunk 11]
```diff
@@ -96,18 +96,43 @@ async function resolveNotificationAssociations({ db, logger, reference, emailAdd
...
+					}
+				}
 			});
+
 			if (app && app.Lpa?.email === emailAddress) {
 				logger.info(`Successfully linked ${notificationId} to contact`);
 				return { Lpa: { connect: { id: app.Lpa.id } } };
 			}
-			if (app && app.ApplicantContact?.email === emailAddress) {
-				contactId = app.applicantContactId;
```
Comment (pinssbrett) [inline apps/manage/src/app/notify/controller.js:106]: If we are deleting these contact IDs as part of the data migration, message callback logs will get corrupted - do these need to be migrated too?

**Review by pinssbrett — CHANGES_REQUESTED** [review]
Great work - this change covers basically the entire codebase, and you even picked up some extra functionality required for multiple entities (e.g. the notify contact linking).
There's still a few bits to remove, including some newly duplicated tests. We'll also need to make sure that data has been successfully migrated for all prod cases (e.g. what if a notify message callback entry is linked to a contact ID that's going to be deleted?)

File: packages/database/src/schema.prisma (hunk 7 above)
Comment (RobBroadhead30) [reply packages/database/src/schema.prisma:62]: I'll speak to Alex on this

File: packages/database/src/schema.prisma (hunk 10 above)
Comment (RobBroadhead30) [reply packages/database/src/schema.prisma:58]: I'll speak to Alex on this

File: apps/manage/src/app/notify/controller.js (hunk 11 above)
Comment (RobBroadhead30) [reply apps/manage/src/app/notify/controller.js:106]: I'll speak to Alex on this

Comment (RobBroadhead30) [reply apps/manage/src/app/notify/controller.js:106]: Alex is in the process of sorting the data that's already in prod. We are okay to merge into test once code is ready

File: packages/database/src/schema.prisma (hunk 10 above)
Comment (RobBroadhead30) [reply packages/database/src/schema.prisma:58]: Alex is in the process of sorting the data that's already in prod. We are okay to merge into test once code is ready

File: packages/database/src/schema.prisma (hunk 7 above)
Comment (RobBroadhead30) [reply packages/database/src/schema.prisma:62]: Alex is in the process of sorting the data that's already in prod. We are okay to merge into test once code is ready

File: apps/manage/src/app/views/cases/create-a-case/journey.test.js (hunk 6 above)
Comment (pinssbrett) [reply apps/manage/src/app/views/cases/create-a-case/journey.test.js]: This is still outstanding - the `describe` statement on line 78 duplicates that on line 8, and you have three duplicate tests:
- `all questions should be defined`: lines 9–21 / lines 82–94
- `should not display secondaryLocalPlanningAuthority when hasSecondaryLpa is NO` lines 54–64 / lines 96–106
- `should display secondaryLocalPlanningAuthority when hasSecondaryLpa is YES` lines 66–76 / lines 108–118

File: apps/manage/src/app/views/cases/view/publish/controller.test.js
Comment (pinssbrett) [reply apps/manage/src/app/views/cases/view/publish/controller.test.js]: Still outstanding - you can remove this entire test as it duplicates that on lines 62-82

File: apps/manage/src/app/views/cases/view/update-case.test.js (hunk 8 above)
Comment (pinssbrett) [reply apps/manage/src/app/views/cases/view/update-case.test.js]: Check the entire file for these keys, they are still present in other tests

File: packages/database/src/schema.prisma (hunk 10 above)
Comment (pinssbrett) [reply packages/database/src/schema.prisma:58]: This is still outstanding - you need to remove these keys from the listed tests

**Review by pinssbrett — CHANGES_REQUESTED** [review]
There's still a few things from the last review that need to be addressed

**Review by pinssbrett — APPROVED** [review]
Great, nice and tidy!
